### PR TITLE
338747: Adding Open API spec for Economic Operators

### PR DESCRIPTION
**Pull request by sean-treanor**

### CODE REVIEW

File: common.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+      description: Bad request
+      content:
+        application/json:
+          schema:
+            type: object
+            properties:
+              error:
+                type: string
+                example: Bad request. Please provide valid parameters.
+    Unauthorized:
```
Comment (sean-treanor) [inline common.yaml]: This file will be removed in a rebase once the branch I copied it from is merged.

**Review by reece-bennett — APPROVED** [review]

File: economic-operator.yaml [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+          type: string
+          description: Identifier for the user organisation
+        userId:
+          type: string
+          description: Identifier for the user
+        telephone:
+          type: string
+          description: Telephone number of the operator
+        effectiveFromDate:
+          type: string
```
Comment (reece-bennett) [inline economic-operator.yaml:99]: The dates can have `format: date` as well

File: economic-operator.yaml [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+          description: Start date of the validity period for the record
+        effectiveToDate:
+          type: string
+          description: End date of the validity period for the record
+        isActive:
+          type: string
```
Comment (reece-bennett) [inline economic-operator.yaml]: `type: boolean`

File: economic-operator.yaml [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+          description: Is the record active
+        createdBy:
+          type: string
+          description: User who created the record
+        createdDate:
+          type: string
```
Comment (reece-bennett) [inline economic-operator.yaml:113]: The timestamps can have `format: date-time`

File: economic-operator.yaml [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+      description: Get all Economic Operators
+      responses: 
+        200:
+          description: Successful response
+          content: 
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: "#/components/schemas/EconomicOperator"
```
Comment (reece-bennett) [inline economic-operator.yaml:18]: Add the error responses too:

```yaml
400:
  $ref: "./common.yaml#/components/responses/BadRequest"
401:
  $ref: "./common.yaml#/components/responses/Unauthorized"
403:
  $ref: "./common.yaml#/components/responses/Forbidden"
```

**Review by reece-bennett — CHANGES_REQUESTED** [review]

**Review by reece-bennett — DISMISSED** [review]

File: economic-operator.yaml [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+        400:
+          $ref: "./common.yaml#/components/responses/BadRequest"
+        401:
+          $ref: "./common.yaml#/components/responses/Unauthorized"
+        403:
+          $ref: "./common.yaml#/components/responses/Forbidden"
+  /economic-operators/{id}:
```
Comment (subash-lakshmikanthan) [inline economic-operator.yaml]: can we please keep just only get all request ?

**Review by kamilmkainos — DISMISSED** [review]

**Review by reece-bennett — DISMISSED** [review]

**Review by kamilmkainos — DISMISSED** [review]

**Review by kamilmkainos — APPROVED** [review]

**Review by reece-bennett — APPROVED** [review]

**Review by subash-lakshmikanthan — APPROVED** [review]

**Pull request merged**